x/confidentialtransfers/types/msgs_test.go

```diff
@@ -0,0 +1,118 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-cryptography/pkg/encryption/elgamal"
+	"github.com/sei-protocol/sei-cryptography/pkg/zkproofs"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestMsgTransfer_FromProto(t *testing.T) {
+	testDenon := "factory/sei1ft98au55a24vnu9tvd92cz09pzcfqkm5vlx99w/TEST"
+	sourcePrivateKey, _ := elgamal.GenerateKey()
+	destPrivateKey, _ := elgamal.GenerateKey()
+	eg := elgamal.NewTwistedElgamal()
+	sourceKeypair, _ := eg.KeyGen(*sourcePrivateKey, testDenon)
+	destinationKeypair, _ := eg.KeyGen(*destPrivateKey, testDenon)
+
+	amountLo := uint64(100)
+	amountHi := uint64(0)
+
+	remainingBalance := uint64(200)
+
+	// Encrypt the amount using source and destination public keys
```

Comment:
Would it make sense to make this it's on function?

func ToProto(transfer *Transfer) *MsgTransfer {}